### ISSUE TITLE
The tool stoped in 2021

### ISSUE THREAD

**Issue by PedroBatista08** [glutanimate/review-heatmap#105]
#### Problem description

*Please describe the issue concisely in here. In case of an error: Walk us through the steps you took to get there. What happened? What did you expect to happen?*

The tool is awesome, but when started 2021 the map stoped to count my reviews and stoped to show the forecast reviews

#### Checklist

*Please replace the space inside the brackets with an **x** if the following items apply:*

 - [ x ] I've restarted Anki to see if it helps
 - [ x ] I've verified that I use the latest version of the add-on by redownloading it from AnkiWeb
 - [ x ] I've verified that I use the latest version of Anki by checking at https://apps.ankiweb.net#download
 - [ x ] I've tried to disable other add-ons to see if there are any interactions present
 - [ ] My issue disappears when I hold shift while starting Anki.
 - [ ] I've checked if anyone else reported this problem before by looking through the issue reports. I also checked to see if there is a section about known issues in the add-on description, documentation, or README.
 

#### Information about your Anki set-up

*Please open Anki, go to Help → About, click on "Copy Debug Info", and paste the result between the backticks below (if the button does not appear you are using an older version of Anki 2.1 and will need to update first):*

```
    
    Anki 2.1.35 (84dcaa86) Python 3.8.0 Qt 5.14.2 PyQt 5.14.2
    Platform: Windows 10
    Flags: frz=True ao=True sv=1
    Add-ons, last update check: 2021-01-02 16:54:18
    
    
    ===Add-ons (active)===
    (add-on provided name [Add-on folder, installed at, version, is config changed])
    Anki Zoom ['538879081', 2020-04-03T21:20, 'v1.1.3', mod]
    AwesomeTTS Google Cloud Text-to-Speech unofficial ['814349176', 2020-10-20T07:40, 'None', '']
    ImageResizer ['1214357311', 2020-12-18T11:01, 'None', '']
    Large and Colorful Buttons ['1829090218', 2018-10-02T14:41, 'None', '']
    Review Heatmap ['review_heatmap', 2020-04-30T10:23, 'None', '']
    True Retention ['613684242', 2017-11-19T17:43, 'None', '']
    
    ===IDs of active AnkiWeb add-ons===
    1214357311 1829090218 538879081 613684242 814349176
    
    ===Add-ons (inactive)===
    (add-on provided name [Add-on folder, installed at, version, is config changed])
    

```

#### Error message (if any)

*If you've received an error message, please copy and paste it between the backticks below:*

```python

```

**Comment by minatoayanokoji** [comment]
It seems I have the same problem as you. When I see the heatmap in continuous timeline, It doesn't show my reviews in this year. Even though It do show in the yearly overview.
Continuous timeline:
![image](https://user-images.githubusercontent.com/76009172/103505622-7a7c4900-4e39-11eb-9af0-ff6230c63822.png)
Yearly overview:
![image](https://user-images.githubusercontent.com/76009172/103505636-8405b100-4e39-11eb-92cc-9ef94bfac2a4.png)

**Comment by henricot7** [comment]
Also have the exact same issue! In Yearly overview it is fine, but in Continuous timeline it isn't working in 2021.

**Comment by warrenalphonso** [comment]
Reinstalling [Version 1.0.0-beta.1](https://github.com/glutanimate/review-heatmap/releases/tag/v1.0.0-beta.1) worked for me. I'm on Anki 2.1.35.

**Comment by warrenalphonso** [comment]
Check out #70. My problem was an invalid due date with a year in the 4 millions.

**Comment by JhonnySalles** [comment]
I have the same problem, I reinstalled for the new version of the extension, but it remains the same.

I noticed that there were some updates on the ankidroid, have the possibility that the new year update broke the extension??

I believe that my mistake should not be in the colors, because with the mouse over the day it shows that there was no data on the day.

![image](https://user-images.githubusercontent.com/43912638/104811972-d2259780-57dd-11eb-99bb-b9d160c789ec.png)

**Comment by Cashburner** [comment]
Same problem here, clicking on tiles seems to solve it. Hope it gets fixed!

**Comment by Cashburner** [comment]
> Reinstalling [Version 1.0.0-beta.1](https://github.com/glutanimate/review-heatmap/releases/tag/v1.0.0-beta.1) worked for me. I'm on Anki 2.1.35.
> 
> Edit: Nevermind. I've got to click on a tile in the heatmap every time I load Anki for the colors to display.

thanks for the "solution"

**Comment by warrenalphonso** [comment]
@JhonnySalles and @Cashburner , did you see my follow-up comment to read issue #70? In prticular, [this comment by Glutanimate](https://github.com/glutanimate/review-heatmap/issues/70#issuecomment-657229570) was the issue for me.

**Comment by lucasgass** [comment]
> Check out #70. My problem was an invalid due date with a year in the 4 millions.
Thanks for posting your solution here :)
Unfortunately, this didn't solve the bug of this thread for me...
I'll keep my eye on here to see if someone has an idea of what's the problem!

**Comment by Cashburner** [comment]
> @JhonnySalles and @Cashburner , did you see my follow-up comment to read issue #70? In prticular, [this comment by Glutanimate](https://github.com/glutanimate/review-heatmap/issues/70#issuecomment-657229570) was the issue for me.

thanks a lot for the solution. a corrupted card caused the issue for me.

**Comment by EduardoMaiaUFMG** [comment]
I am having the same issue. I did 372 reviews today, but it only shows 72. In addition, I have around 350 cards for tomorrow, but it shows none. The previous and following days are normal though.

**Comment by howl1ng** [comment]
> I am having the same issue. I did 372 reviews today, but it only shows 72. In addition, I have around 350 cards for tomorrow, but it shows none. The previous and following days are normal though.

Yeah, I'm having a similar issue. I reviewed 105 cards today, but the add-on shows only 31 reviewed cards. It shows no cards due tomorrow either, but when I click on "Browse" I can see there are indeed some cards due tomorrow, even if the add-on doesn't show it.

I also have the issue of the continuous timeline not working, but seeing as I generally use the yearly review style, I hadn't noticed the issue before.

Did anyone find a solution to the problem?

**Comment by EduardoMaiaUFMG** [comment]
> > I am having the same issue. I did 372 reviews today, but it only shows 72. In addition, I have around 350 cards for tomorrow, but it shows none. The previous and following days are normal though.
> 
> Yeah, I'm having a similar issue. I reviewed 105 cards today, but the add-on shows only 31 reviewed cards. It shows no cards due tomorrow either, but when I click on "Browse" I can see there are indeed some cards due tomorrow, even if the add-on doesn't show it.
> 
> I also have the issue of the continuous timeline not working, but seeing as I generally use the yearly review style, I hadn't noticed the issue before.
> 
> Did anyone find a solution to the problem?

are you from Brazil by any chance? because I think it might be due to the daylight saving time that we were supposed to have here tomorrow. maybe the time setting is all messed up because of this. didn't see anyone complaining about it on reddit ...

**Comment by willycamargo** [comment]
> are you from Brazil by any chance? because I think it might be due to the daylight saving time that we were supposed to have here tomorrow. maybe the time setting is all messed up because of this. didn't see anyone complaining about it on reddit ...

I'm from Brazil and come here to see if someone has tracked this bug. I'll wait until monday to see what happens with the chart and maybe work on a fork to solve this issue.

**Comment by howl1ng** [comment]
> > > I am having the same issue. I did 372 reviews today, but it only shows 72. In addition, I have around 350 cards for tomorrow, but it shows none. The previous and following days are normal though.
> > 
> > 
> > Yeah, I'm having a similar issue. I reviewed 105 cards today, but the add-on shows only 31 reviewed cards. It shows no cards due tomorrow either, but when I click on "Browse" I can see there are indeed some cards due tomorrow, even if the add-on doesn't show it.
> > I also have the issue of the continuous timeline not working, but seeing as I generally use the yearly review style, I hadn't noticed the issue before.
> > Did anyone find a solution to the problem?
> 
> are you from Brazil by any chance? because I think it might be due to the daylight saving time that we were supposed to have here tomorrow. maybe the time setting is all messed up because of this. didn't see anyone complaining about it on reddit ...

I am, actually! Did my writing give it away? Haha

Yeah, I don't know about daylight saving time, because that's been cancelled since 2019 or 2020, and the add-on was working fine then

**Comment by EduardoMaiaUFMG** [comment]
> > > > I am having the same issue. I did 372 reviews today, but it only shows 72. In addition, I have around 350 cards for tomorrow, but it shows none. The previous and following days are normal though.
> > > 
> > > 
> > > Yeah, I'm having a similar issue. I reviewed 105 cards today, but the add-on shows only 31 reviewed cards. It shows no cards due tomorrow either, but when I click on "Browse" I can see there are indeed some cards due tomorrow, even if the add-on doesn't show it.
> > > I also have the issue of the continuous timeline not working, but seeing as I generally use the yearly review style, I hadn't noticed the issue before.
> > > Did anyone find a solution to the problem?
> > 
> > 
> > are you from Brazil by any chance? because I think it might be due to the daylight saving time that we were supposed to have here tomorrow. maybe the time setting is all messed up because of this. didn't see anyone complaining about it on reddit ...
> 
> I am, actually! Did my writing give it away? Haha
> 
> Yeah, I don't know about daylight saving time, because that's been cancelled since 2019 or 2020, and the add-on was working fine then
> 
> The lack of complaining on reddit is pretty telling as well, maybe it's only happening in Brazil?

noo not at all hahaha it's just because this same thing happened to me last year in november (when daylight-saving-time changes were supposed to start) and right now (when it was supposed to finish). But I hope the reviews will show up tomorrow :/ I like the styling of the heatmap on the desktop haha

**Comment by Cashburner** [comment]
I am from Greece and I got the same issue in the past. Not quite sure what caused it. Can you maybe tell me if 1. you did reviews from phone and synced them to your pc 2. exported/imported any deck(s) 3. have an addon for auto ease or auto bury or free weekend/load balance? I am trying to find where these issues are coming from and I think they might have sth to do with these

**Comment by gauerin** [comment]
Greetings from Brazil! I'm actually having this problem too! 

![image](https://user-images.githubusercontent.com/71511259/108629225-bb5a0c80-743d-11eb-846d-60ff6bad19bd.png)
![image](https://user-images.githubusercontent.com/71511259/108629243-cad95580-743d-11eb-843a-3d8f998f2124.png)

I already tried to reinstall all versions of the program and the addon, but the issue remains.

**Comment by EduardoMaiaUFMG** [comment]
> I am from Greece and I got the same issue in the past. Not quite sure what caused it. Can you maybe tell me if 1. you did reviews from phone and synced them to your pc 2. exported/imported any deck(s) 3. have an addon for auto ease or auto bury or free weekend/load balance? I am trying to find where these issues are coming from and I think they might have sth to do with these

1. yes / 2. yes - but it was already happening b4 that / 3. nop
I'm not the author of the post but i'm having the same issue

**Comment by Bhar0kan** [comment]
> > I am having the same issue. I did 372 reviews today, but it only shows 72. In addition, I have around 350 cards for tomorrow, but it shows none. The previous and following days are normal though.
> 
> Yeah, I'm having a similar issue. I reviewed 105 cards today, but the add-on shows only 31 reviewed cards. It shows no cards due tomorrow either, but when I click on "Browse" I can see there are indeed some cards due tomorrow, even if the add-on doesn't show it.
> 
> I also have the issue of the continuous timeline not working, but seeing as I generally use the yearly review style, I hadn't noticed the issue before.
> 
> Did anyone find a solution to the problem?

I have exactly the same problem

**Comment by JhonnySalles** [comment]
I am from Brazil too. I've been analyzing the cards, but I haven't found any corrupted cards as previously reported.

I don't really understand how it was designed, but my mistake started to occur on January 1st, and the summer time should have started before that.
I do not say that this evidence can prove that it has nothing related, but I think it is a little strange in my case. I will try to activate daylight saving time manually and check if it returns to the normal state.

**Comment by lucasgass** [comment]
So it's monday and: the bug of saturday, where a smaller number of cards done were shown, is now fixed.
The bug of sunday, where no cards done are being shown, is still there.
And the main bug of this thread, which is that in "continuous timeline" mode no cards done are shown in 2021, is still there and (I believe) unrelated to this sunday's bugs.

**Comment by EduardoMaiaUFMG** [comment]
> So it's monday and: the bug of saturday, where a smaller number of cards done were shown, is now fixed.
> The bug of sunday, where no cards done are being shown, is still there.
> And the main bug of this thread, which is that in "continuous timeline" mode no cards done are shown in 2021, is still there.
> 
> Btw: I'm also from BR and it seems that this sunday's bug was related to our summer daylight time, but the 2021 problem is for a whole different reason.

yeah I've seen some people from different countries complaining about the continuous timeline too. regarding the other problems, my heatmap kind of summed up the number of reviews that I did on the weekend. thus, it shows that I did 778 reviews on saturday and 0 on sunday. weird

**Comment by mds-2021** [comment]
As a temporary workaround, setting the timezone to Buenos Aires (what I did) or setting the time manually, as mentioned above, on the Windows 10 Settings page solves both the 'February 21 blank spot' and the 'Continuous Timeline 2021 year blank' problems.

**Comment by willycamargo** [comment]
For those that are from Brazil and are having problems with the date February 21st. I installed the version [0.7.0-beta.1](https://github.com/glutanimate/review-heatmap/releases/tag/v0.7.0-beta.1) and it's working fine.

**Comment by EduardoMaiaUFMG** [comment]
> For those that are from Brazil and are having problems with the date February 21st. I installed the version [0.7.0-beta.1](https://github.com/glutanimate/review-heatmap/releases/tag/v0.7.0-beta.1) and it's working fine.

tks !! now everything's fixed. I appreciate it :)

**Comment by JhonnySalles** [comment]
for me decided in parts to update.
I saw that when resetting the settings it returned to work correctly, but when changing the type of calendar it stops in 2021 as the images below.
Maybe it is not something in the cards or time zone, but in the type of calendar used, because normal shows correctly, only continuous shows wrong.

![image](https://user-images.githubusercontent.com/43912638/109172550-ac3acd80-7761-11eb-8c79-e17df0f06cfc.png)


![image](https://user-images.githubusercontent.com/43912638/109172585-b4930880-7761-11eb-9140-9c4d11060cc9.png)

**Comment by PHZCosta** [comment]
Brazilian over here also. 
I wasn't having this problem but then updated from 2.1.34 to 2.1.40 and this problem started. Changing from continuous time to the yearly overview "solves" it, but now I have a hole in 21 February (although it doesn't mess up with the streak)
